dotCMS/src/main/java/com/dotmarketing/portlets/variant/business/VariantAPIImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.dotmarketing.portlets.variant.business;
+
+import com.dotcms.business.WrapInTransaction;
+import com.dotcms.rest.validation.Preconditions;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.variant.model.Variant;
+import com.dotmarketing.util.Logger;
+import java.util.Optional;
+
+/**
+ * API of {@link Variant}
+ */
+public class VariantAPIImpl implements VariantAPI {
+    private VariantFactory variantFactory;
+
+    public VariantAPIImpl(){
+        variantFactory = FactoryLocator.getVariantFactory();
+    }
+
+    /**
+     * Implementation for {@link VariantAPI#save(Variant)}
+     * @param variant
+     *
+     * @return
+     * @throws DotDataException
+     */
+    @Override
+    @WrapInTransaction
+    public Variant save(final Variant variant) {
+
+        Preconditions.checkNotNull(variant.getName(), "Variant name should not be null");
+
+        if (variant.isArchived()) {
```

Comment:
This can be a `DotPreconditions` call also

Reply:
done https://github.com/dotCMS/core/pull/22839/files#diff-a2746c1b6a4b623f65d595e778def0a3d7ce80f05cf45d412a4472b1e6b1bf92R36